state-sync/state-sync-driver/src/storage_synchronizer.rs

```diff
@@ -799,13 +802,21 @@ fn spawn_commit_post_processor<
 fn write_kv_to_indexer_db(
     internal_indexer_db: &Option<Arc<DB>>,
     kvs: &Vec<(StateKey, StateValue)>,
+    snapshot_version: Version,
+    write_version: bool,
```

Comment:
hmm.. just realized this code should be shared between fast sync and db-restore, so either add it to StateRestore or the indexer folder.